pkg/api/customization/multiclusterapp/store.go

```diff
@@ -105,6 +105,13 @@ func (s *Store) setDisplayName(apiContext *types.APIContext, schema *types.Schem
 			if err != nil {
 				return nil, err
 			}
+			if token.AuthProvider == providers.LocalProvider {
+				// member's principal id is from an external auth provider, but caller is logged in as local user,
+				// getPrincipal if called, will be called on local auth provider, and will not find a user with this principal ID
+				// hence returning, since the only thing this method does is setting display name by getting it from the external auth
+				// provider.
+				return data, nil
+			}
```

Comment:
Will we ever need to setDisplayName when  token.AuthProvider == providers.LocalProvider? If we can call this logic only when it is some external auth provider login, can we maybe make this check at the start of this method?

Reply:
Yup, missed this earlier, this check can be done once outside the for loop